### ISSUE TITLE
Running CAEVO-plus on raw documents

### ISSUE THREAD

**Issue by svjan5** [forkunited/CAEVO-plus#1]
Hi,
Can you list the procedure or provide some bash script for running CAEVO-plus on a directory or file, as it was provided for the earlier version of CAEVO?

Thanks in advance
Shikhar

**Comment by forkunited** [comment]
Hi Shikhar,

The system isn't currently setup to run on raw text through a single command.  The text would need to be pre-processed and annotated first.   I haven't had time to setup an end-to-end program for this version.

If you're just trying to produce ordering annotations for your text, the original CAEVO system should probably be sufficient.  The performance gains of this newer version might not be so large that there would be much difference with respect to using the systems for practical purposes.  This is especially true if your text differs substantially from the text in the TimeBank-Dense corpus on which the system was trained and evaluated.

I can write a script for processing raw text at some point in the future.  It may be a while before I get to it, though, so it would probably be best to use the original CAEVO for now, if you can.

Bill

**Comment by svjan5** [comment]
Hi,
Thanks for the prompt reply. 
Sure, as of now I will stick to the previous version of CAEVO but it would be great if you could provide the preprocessing script at the earliest.

Thanks
Shikhar

**Comment by vedmathai** [comment]
@forkunited 
Even if you don't have the time to create a script that can run CAEVO-PLUS on raw files, do you think you'd be able to update the README (or another doc) with steps, or pointers on how to go about processing raw text etc like @svjan5 asked above (as opposed to writing a script for it).
I am hoping if you can point me at the preprocessors and other things, I can try putting them together in a script.
I want to report on top of the 5-8% gain CAEVO-PLUS got over the original.

Ved

**Comment by forkunited** [comment]
@vedmathai 

Sorry I never got around to doing the raw text thing (I forgot someone had requested this last year).  I might have some time to get it done this week.  However, if you're trying to report on top of the gain of CAEVO-plus over the original, that shouldn't require running on raw text.  The numbers in both the original CAEVO and CAEVO plus papers are reported for predicting tlinks over a fixed set of timexes and events in TimeBank dense (i.e. none of the reported results relies on predicting the timexes and events prior to the tlinks).  Maybe I misunderstand what you're trying to do though?

**Comment by vedmathai** [comment]
I am trying to club the concept of cross-doc event coreference along with dense event relations to create even denser tlinks across documents.
I am trying to use the gold standard ECB+ dataset for the coreference parts. So I thought I could use this CAEVO plus code to tag dense tlinks, since ECB+ doesn't have that.
You can probably help me: You had alluded to using event coreferences in the 'future work' part of your paper. Have you or do you know any work that has already implemented this?

**Comment by forkunited** [comment]
Ah I think I actually tried something similar, but with ACE2005 a few years ago.  I had a sieve that did both event coref and temporal ordering jointly (and cross-document, I think), but I didn't end up with any significant results, so didn't end up publishing.    The actual scripts for it are in this repo that CAEVO-plus is actually based on (https://github.com/forkunited/micro-event). But I don't recommend trying to use them---they are undocumented, confusing, and unnecessarily complicated.  

I'll still attempt getting CAEVO-plus setup on raw text sometime soon (hopefully this week).  Sorry for the delay.  I also noticed that I had forgotten to add the scripts necessary for reproducing the results to the repo, so the current run.sh script is broken.  So thanks for inadvertently helping me to notice that.  I'll add those too.

**Comment by forkunited** [comment]
@vedmathai and @svjan5 

I just added an option for running on a raw text file.  Hopefully that should work, but let me know if you run into any weird issues or bugs, or if you have any questions.  It's a bit hacked together, so sorry in advance for that.  As I mentioned above, for many purposes, it may also just be sufficient to resort to the original CAEVO system if the CAEVO-plus version is too clunky.

Also, sorry @svjan5 that it took me more than a year to get around to this---I realized you've probably moved on by now.  I was very busy when you asked, and I hoped it would probably be fine for you to use the original CAEVO, and then I totally forgot about getting around to it.
